lib/Backends/Backend.cpp

```diff
@@ -71,27 +71,65 @@ void Backend::autoInstrument(TraceInfo &traceInfo, IRFunction *IR) const {
   // Get all instructions in the IRFunction.
   IRFunction::InstListTy &instructions = IR->getInstrs();
 
-  // Build a placeholder to a backing tensor with space to fit all timestamps.
-  auto *backingPH = F->getParent()->createPlaceholder(
+  // First pass, find out how many TraceEvents we should add. Existing
+  // TraceEvents have their own backing Tensors, so don't count them.
+  size_t numEvents = 1; // Starts at 1 since there is always a start event.
+  for (auto it = instructions.begin(); it != instructions.end(); it++) {
+    auto &I = *it;
+    bool isInstrumentation = llvm::isa<TraceEventInst>(&I);
+    if (!isInstrumentation) {
+      numEvents++;
+    }
+  }
+
+  // Default name for the instrumentation placeholder, will be made unique by
+  // createPlaceholder.
+  std::string name = F->getName().str() + "_insrumentation";
+  Placeholder *backingPH = F->getParent()->getPlaceholderByName(name);
+
+  auto &varmap = IR->getVariableMap();
+  auto type = F->getParent()->uniqueType(
       ElemKind::Int64ITy,
-      {instructions.size() + 1,
-       getTraceEventDataSize() / Type::getElementSize(ElemKind::Int64ITy)},
-      F->getName().str() + "_instrumentation", false);
+      {numEvents,
+       getTraceEventDataSize() / Type::getElementSize(ElemKind::Int64ITy)});
+
+  WeightVar *backingWeight = nullptr;
```

Comment:
Can you add some comments here about what's going on for the different combinations here (`backingWeight` true/false, and `backingPH` true/false).